drivers/modem/hl7800.c

```diff
@@ -1783,8 +1783,11 @@ static bool on_cmd_atcmdinfo_iccid(struct net_buf **buf, uint16_t len)
 		LOG_INF("EID: %s", delim + 1);
 	}
 
-	ret = snprintk(iface_ctx.mdm_iccid, sizeof(iface_ctx.mdm_iccid), "%s", value);
-	if (ret > MDM_HL7800_ICCID_MAX_STRLEN) {
+	out_len = MIN(sizeof(iface_ctx.mdm_iccid) - 1, strlen(value));
```

Comment:
I might have written this as
```
ret = strlen(value);
strncpy(iface_ctx.mdm_iccid, value, sizeof(iface_ctx.mdm_iccid));
iface_ctx.mdm_iccid[sizeof(iface_ctx.mdm_iccid)-1] = '\0';

if (ret > MDM_HL7800_ICCID_MAX_STRLEN) {
```
That would preserve the surrounding logic and left the 'ret' value with the expected contents, otherwise I'm not sure what 'ret' contains.

Reply:
Agree, the LOG_WRN below will print `ret` that is no longer used.

I think this is what is needed:
```
ret = strlen(value);
strncpy(iface_ctx.mdm_iccid, value, sizeof(iface_ctx.mdm_iccid));
if (ret > (sizeof(iface_ctx.mdm_iccid)-1) {
    ret = sizeof(iface_ctx.mdm_iccid)-1;
}
iface_ctx.mdm_iccid[ret] = '\0';

if (ret > MDM_HL7800_ICCID_MAX_STRLEN) {
```

Reply:
I see, ok went with this version.